x/harvest/keeper/liquidation.go

```diff
@@ -276,6 +244,55 @@ func (k Keeper) StartAuctions(ctx sdk.Context, borrower sdk.AccAddress, borrows,
 	return nil
 }
 
+// IsWithinValidLtvRange compares a borrow and deposit to see if it's within a valid LTV range at current prices
+func (k Keeper) IsWithinValidLtvRange(ctx sdk.Context, deposit types.Deposit, borrow types.Borrow) (bool, error) {
+	liqMap, err := k.LoadLiquidationData(ctx, deposit, borrow)
+	if err != nil {
+		return false, err
+	}
+
+	totalBorrowableUSDAmount := sdk.ZeroDec()
+	totalDepositedUSDAmount := sdk.ZeroDec()
+	for _, depCoin := range deposit.Amount {
+		lData := liqMap[depCoin.Denom]
+		usdValue := sdk.NewDecFromInt(depCoin.Amount).Quo(sdk.NewDecFromInt(lData.conversionFactor)).Mul(lData.price)
+		totalDepositedUSDAmount = totalDepositedUSDAmount.Add(usdValue)
+		borrowableUSDAmountForDeposit := usdValue.Mul(lData.ltv)
+		totalBorrowableUSDAmount = totalBorrowableUSDAmount.Add(borrowableUSDAmountForDeposit)
+	}
+
+	totalBorrowedUSDAmount := sdk.ZeroDec()
+	for _, coin := range borrow.Amount {
+		lData := liqMap[coin.Denom]
+		usdValue := sdk.NewDecFromInt(coin.Amount).Quo(sdk.NewDecFromInt(lData.conversionFactor)).Mul(lData.price)
+		totalBorrowedUSDAmount = totalBorrowedUSDAmount.Add(usdValue)
+	}
+
+	// Check if the user's has borrowed more than they're allowed to
+	if totalBorrowedUSDAmount.GT(totalBorrowableUSDAmount) {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// UpdateItemInLtvIndex updates the key a borrower's address is stored under in the LTV index
+func (k Keeper) UpdateItemInLtvIndex(ctx sdk.Context, prevLtv sdk.Dec,
+	shouldRemoveIndex bool, borrower sdk.AccAddress) error {
+	currLtv, shouldInsertIndex, err := k.GetStoreLTV(ctx, borrower)
+	if err != nil {
+		return err
+	}
+
+	if shouldRemoveIndex {
+		k.RemoveFromLtvIndex(ctx, prevLtv, borrower)
+	}
+	if shouldInsertIndex {
+		k.InsertIntoLtvIndex(ctx, currLtv, borrower)
+	}
+	return nil
+}
+
 // GetStoreLTV calculates the user's current LTV based on their deposits/borrows in the store
 // and does not include any outsanding interest.
 func (k Keeper) GetStoreLTV(ctx sdk.Context, addr sdk.AccAddress) (sdk.Dec, bool, error) {
```

Comment:
Update function comment to describe what the boolean return variable signifies.

Reply:
https://github.com/Kava-Labs/kava/pull/747/commits/4d78ea83c3a0b2ba4ecafb0087d1c07a364a5f81